src/main/java/emu/grasscutter/game/player/Player.java

```diff
@@ -797,6 +800,43 @@ public void addAvatar(int avatarId) {
         addAvatar(new Avatar(avatarId), true);
     }
 
+    public boolean addTrialAvatarForQuest(int trialAvatarId, int questMainId){
+        // TODO, other trial avatar like activity and element trial dungeon might have 
+        // completely different scenario, this function is currently used for Quest Exec only
+        TrialAvatarData trialAvatar = GameData.getTrialAvatarDataMap().get(trialAvatarId);
+        Avatar avatar = new Avatar(trialAvatar.getTrialAvatarParamList().get(0));
+        if (avatar.getAvatarData() == null || !hasSentLoginPackets()) {
+            return false;
+        }
+        avatar.setOwner(this);
+        // Add trial weapons and relics
+        avatar.setTrialAvatarInfo(trialAvatar, GrantReason.GRANT_REASON_BY_QUEST, questMainId);
+        avatar.equipTrialItems();
+        // Recalc stats
+        avatar.recalcStats();
+
+        // Packet, mimic official server behaviour, add to player's bag but not saving to db
+        sendPacket(new PacketAvatarAddNotify(avatar, false));
+        // add to avatar to temporary trial team
+        getTeamManager().addAvatarToTrialTeam(avatar);
+        // Packet, mimic official server behaviour, neccessary to stop player from modifying team 
+        sendPacket(new PacketAvatarTeamUpdateNotify(this));
+        return true;
+    }
+
+    public boolean removeTrialAvatar(){
+        if (getTeamManager().getTrialTeamGuid() == null || getTeamManager().getTrialTeamGuid().size() == 0) {
```

Comment:
shouldn't this have a parameter? the execs I saw seem to direclty remove specific trial characters, not all.